src/plugins/chart_expressions/expression_xy/common/expression_functions/reference_line.ts

```diff
@@ -0,0 +1,114 @@
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  AvailableReferenceLineIcons,
+  FillStyles,
+  IconPositions,
+  LayerTypes,
+  LineStyles,
+  REFERENCE_LINE,
+  REFERENCE_LINE_Y_CONFIG,
+  YAxisModes,
+} from '../constants';
+import { ReferenceLineFn } from '../types';
+import { strings } from '../i18n';
+
+export const referenceLineFunction: ReferenceLineFn = {
+  name: REFERENCE_LINE,
+  aliases: [],
+  type: REFERENCE_LINE,
+  help: strings.getRLHelp(),
+  inputTypes: ['datatable', 'null'],
+  args: {
+    name: {
+      types: ['string'],
+      help: strings.getReferenceLineNameHelp(),
+    },
+    value: {
+      types: ['number'],
+      help: strings.getReferenceLineValueHelp(),
+      required: true,
+    },
+    axisMode: {
+      types: ['string'],
+      options: [...Object.values(YAxisModes)],
+      help: strings.getAxisModeHelp(),
+      default: YAxisModes.AUTO,
+      strict: true,
+    },
+    color: {
+      types: ['string'],
+      help: strings.getColorHelp(),
+    },
+    lineStyle: {
+      types: ['string'],
+      options: [...Object.values(LineStyles)],
+      help: i18n.translate('expressionXY.yConfig.lineStyle.help', {
+        defaultMessage: 'The style of the reference line',
+      }),
+      default: LineStyles.SOLID,
+      strict: true,
+    },
+    lineWidth: {
+      types: ['number'],
+      help: i18n.translate('expressionXY.yConfig.lineWidth.help', {
+        defaultMessage: 'The width of the reference line',
+      }),
+      default: 1,
+    },
+    icon: {
+      types: ['string'],
+      help: i18n.translate('expressionXY.yConfig.icon.help', {
+        defaultMessage: 'An optional icon used for reference lines',
+      }),
+      options: [...Object.values(AvailableReferenceLineIcons)],
+      strict: true,
+    },
+    iconPosition: {
+      types: ['string'],
+      options: [...Object.values(IconPositions)],
+      help: i18n.translate('expressionXY.yConfig.iconPosition.help', {
+        defaultMessage: 'The placement of the icon for the reference line',
+      }),
+      default: IconPositions.AUTO,
+      strict: true,
+    },
+    textVisibility: {
+      types: ['boolean'],
+      help: i18n.translate('expressionXY.yConfig.textVisibility.help', {
+        defaultMessage: 'Visibility of the label on the reference line',
+      }),
+    },
+    fill: {
+      types: ['string'],
+      options: [...Object.values(FillStyles)],
+      help: i18n.translate('expressionXY.yConfig.fill.help', {
+        defaultMessage: 'Fill',
+      }),
+      default: FillStyles.NONE,
+      strict: true,
+    },
+  },
+  fn(table, args) {
+    const textVisibility =
+      args.name !== undefined && args.textVisibility === undefined
+        ? true
+        : args.name === undefined
+        ? false
+        : args.textVisibility;
+
+    return {
+      type: REFERENCE_LINE,
+      layerType: LayerTypes.REFERENCELINE,
```

Comment:
Basically what I'm suggesting is to turn this into a `REFERENCE_LINE_LAYER` type and wrapping the value into a `table` argument. Everything downstream should be identical so it can be handled exactly the same way in the renderer code, right?

Reply:
If there's something that's not identical then we should discuss why this is the case and whether it makes sense.